builtin/providers/azurerm/resource_arm_virtual_machine.go

```diff
@@ -1370,18 +1394,35 @@ func expandAzureRmVirtualMachineImageReference(d *schema.ResourceData) (*compute
 	storageImageRefs := d.Get("storage_image_reference").(*schema.Set).List()
 
 	storageImageRef := storageImageRefs[0].(map[string]interface{})
+	imageReference := compute.ImageReference{}
+	imageID := ""
+	publisher := ""
+
+	if storageImageRef["image_id"] != "" {
+		imageID = storageImageRef["image_id"].(string)
+		imageReference = compute.ImageReference{
+			ID: &imageID,
+		}
+	} else {
+		publisher = storageImageRef["publisher"].(string)
+		offer := storageImageRef["offer"].(string)
+		sku := storageImageRef["sku"].(string)
+		version := storageImageRef["version"].(string)
+
+		imageReference = compute.ImageReference{
+			Publisher: &publisher,
+			Offer:     &offer,
+			Sku:       &sku,
+			Version:   &version,
+		}
+	}
 
-	publisher := storageImageRef["publisher"].(string)
-	offer := storageImageRef["offer"].(string)
-	sku := storageImageRef["sku"].(string)
-	version := storageImageRef["version"].(string)
-
-	return &compute.ImageReference{
-		Publisher: &publisher,
-		Offer:     &offer,
-		Sku:       &sku,
-		Version:   &version,
-	}, nil
+	//BEGIN: code to be removed after GH-13016 is merged
+	if imageID != "" && publisher != "" {
```

Comment:
Can this ever be true? You are setting publisher and imageID to "" - then using an if / else to hydrate the struct.